person.rb

```diff
@@ -0,0 +1,34 @@
+require_relative './nameable'
+require_relative './decorator'
+
+class Person < Nameable
+  attr_accessor :name, :age
+  attr_reader :id, :rentals
+
+  def initialize(age, name: 'Unknown', parent_permission: true)
+    super()
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  def correct_name
+    @name
+  end
+
+  def add_rental(date, person)
+    Rental.new(date, person, self)
+  end
```

Comment:
- [ ] The purpose of this method is to allow the creation of a rental given a date and a book, as we already have access to the person through `self`. So this method should be taking `date` and `book`, not date and person.

- [ ] Also the arrangement of arguments here is wrong. You defined your Rental initialize method to expect parameters in this order `date`, `book` and `person`.

  Here is hint on how to fix both issues:
  ```
  def add_rental(date, book)
    Rental.new(date, book, self)
  end
  ```

Reply:
Thank you for your feedback. That method has been updated.